File: app/src/main/java/com/auto/clear/notification/MainActivity.java
package com.auto.clear.notification;

import android.app.Activity;
import android.os.Bundle;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

public final class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int padding = (int) (24 * getResources().getDisplayMetrics().density);
        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setGravity(Gravity.CENTER_VERTICAL);
        root.setPadding(padding, padding, padding, padding);
        root.setLayoutParams(new LinearLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
        ));

        TextView title = new TextView(this);
        title.setText(getString(R.string.app_name));
        title.setTextSize(22);

        TextView body = new TextView(this);
        body.setText("Enable this AP101 Xposed module in LSPosed or Vector, then scope it to the apps whose own non-ongoing notifications should be cleared when they regain focus.");
        body.setTextSize(15);
        body.setPadding(0, padding / 2, 0, 0);

        root.addView(title);
        root.addView(body);
        setContentView(root);
    }
}

File: app/src/main/java/com/auto/clear/notification/MainActivity.kt
package com.auto.clear.notification

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Card
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MaterialTheme {
                Surface(modifier = Modifier.fillMaxSize()) {
                    MainScreen()
                }
            }
        }
    }
}

@Composable
private fun MainScreen() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Text(
            text = "Auto-Clear Notification",
            style = MaterialTheme.typography.headlineMedium
        )
        Text(
            text = "Modern libxposed API 101 module",
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )

        InfoCard(
            title = "Behavior",
            body = "When a scoped app gains window focus, the module clears that app's own non-ongoing notifications. It does not hook system_server and does not try to cancel notifications from unrelated apps."
        )

        InfoCard(
            title = "How to use",
            body = "1. Install the APK.\n2. Enable the module in LSPosed or Vector.\n3. Scope it to the apps you want cleaned.\n4. Reopen the target app.\n5. Ongoing notifications such as media playback are preserved."
        )

        InfoCard(
            title = "Notes",
            body = "This build intentionally removes the old broken settings bridge. The module now exposes one fixed lightweight policy so behavior is deterministic and matches the implementation."
        )
    }
}

@Composable
private fun InfoCard(title: String, body: String) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = body,
                style = MaterialTheme.typography.bodyMedium
            )
        }
    }
}
